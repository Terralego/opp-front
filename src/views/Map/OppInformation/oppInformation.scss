@import "../../../variables";

.opp-information {

  &__overlay {
    display: flex;
    justify-content: center;
    @include rounded(8px);
    
    &.bp3-overlay-scroll-container {
      overflow: hidden;
    }
    
    .bp3-overlay-backdrop {
      &.bp3-overlay-enter-done {
        left: $navbar-width;
      }
    }
  }

  &__trigger {
    position: absolute;
    top: 0.7rem;
    left: 2.5rem;
    background-color: $mapPopup;
    background-image: none;
  
    &:hover {
      text-decoration: none;
      background-color: rgba(71, 93, 109, 0.96);
      color: #ffffff;
    }
  
    > span {
      color: $white;
    }
  }
}

.opp-information-overlay {
  &__title {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background: $activetabgray;
    position: relative;
    @include rounded(8px 8px 0 0);
    border-bottom: 2px solid rgba(16, 22, 26, 0.40);

    button.bp3-button {
      border: none;
      box-shadow: none;
      margin: 1em;
      background-color: transparent;
      background-image: none;
      transition-duration: 0.5s;
      opacity: 0.5;

      &:hover {
        transform: rotate(90deg);
        transition-duration: 0.5s;
        background-color: transparent;
        background-image: none;
        box-shadow: none;
        opacity: 0.8;

      }
    }
  }

  .bp3-overlay-content {
    bottom: 0;
    top: 1280px;
    width: 700px;
    height: 300px;
    color: $white;

    &.bp3-overlay-enter-done {
      top: 150px;
    }

    h2 {
      padding: 0 20px;
      font-family: $base-font;
      font-size: 22px;
      color: #F5F8FA;
      line-height: $base-line-height;
      font-weight: 200;
    }

    .bp3-overlay-subcontent {
      padding: 20px;
      background-color: #223947;
      @include rounded(0 0 8px 8px);

      h3 {
        font-family: $base-font;
        font-size: 18px;
        color: #F5F8FA;
        margin: 0.5em 0;
        font-weight: 200;
      }

      p {
        font-family: $base-font;
        font-size: 14px;
        color: #F5F8FA;
        line-height: $base-line-height;
      }

      .subcontent-img {
        display: flex;
        justify-content: space-between;
        flex-direction: row;
        margin: 2em 0 1em;

        img {
          max-height: 160px;
          padding: 0 1em;
        }
      }
    }
  }
}
