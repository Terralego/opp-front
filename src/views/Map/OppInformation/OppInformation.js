import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Classes, Button, Icon, Overlay, Position, Tooltip } from '@blueprintjs/core';
import { useTranslation } from 'react-i18next';

import classnames from 'classnames';

import './oppInformation.scss';

const isHelpHidden = () => localStorage.getItem('isHelpHidden') === 'true';

export const OppInformation = ({ className, isOpen: defaultOpened }) => {
  const [isOpen, setOpen] = useState(defaultOpened);

  const { t } = useTranslation();

  useEffect(() => {
    !isHelpHidden() && setOpen(true);
  }, []);

  const toggleOpen = () => {
    localStorage.setItem('isHelpHidden', isOpen);
    setOpen(!isOpen);
  };

  return (
    <div className={classnames(Classes.DARK, className, 'opp-information')}>
      <Overlay
        autoFocus
        canEscapeKeyClose
        canOutsideClickClose
        enforceFocus
        hasBackdrop
        usePortal
        onClose={toggleOpen}
        isOpen={isOpen}
        className="opp-information__overlay opp-information-overlay"
      >
        <div>
          <div className="opp-information-overlay__title">
            <h2>Information</h2>
            <Button
              icon={
                <Tooltip content={t('common.close')} position={Position.TOP_LEFT}>
                  <Icon icon="cross" iconSize="34" color="#fff" />
                </Tooltip>
              }
              onClick={toggleOpen}
            />
          </div>
          <div className="opp-information-overlay__content bp3-overlay-subcontent">
            <p>Put your information here</p>
          </div>
        </div>
      </Overlay>
      <Button className="opp-information__trigger" icon="info-sign" onClick={toggleOpen} />
    </div>
  );
};

OppInformation.propTypes = {
  className: PropTypes.string,
  isOpen: PropTypes.bool,
};

OppInformation.defaultProps = {
  className: '',
  isOpen: false,
};

export default OppInformation;
