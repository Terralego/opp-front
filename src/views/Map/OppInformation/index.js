import OppInformation from './OppInformation';

export default OppInformation;
