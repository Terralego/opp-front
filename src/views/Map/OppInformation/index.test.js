import React from 'react';
import renderer from 'react-test-renderer';

import OppInformation from './OppInformation';

it('should render', () => {
  const tree = renderer.create((
    <OppInformation />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
