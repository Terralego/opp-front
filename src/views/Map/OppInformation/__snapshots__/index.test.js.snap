// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="bp3-dark opp-information"
>
  <button
    className="bp3-button opp-information__trigger"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    type="button"
  >
    <span
      className="bp3-icon bp3-icon-info-sign"
      icon="info-sign"
    >
      <svg
        data-icon="info-sign"
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <desc>
          info-sign
        </desc>
        <path
          d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zM7 3h2v2H7V3zm3 10H6v-1h1V7H6V6h3v6h1v1z"
          fillRule="evenodd"
        />
      </svg>
    </span>
  </button>
</div>
`;
